agreement_legal_sale_fieldservice/models/sale_order.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _action_confirm(self):
+        res = super(SaleOrder, self)._action_confirm()
+        for order in self:
+            agreement_id = self.env['agreement'].\
+                search([('code', '=', order.name)])
+            agreement_id.fsm_location_id = order.partner_id.\
+                service_location_id
```

Comment:
@osi-scampbell What about using the FSM location of the order?

Reply:
@max3903 This module only depends on agreement_legal_sale and fieldservice_agreement, I can implement that workflow if I add fieldservice_sale to the dependencies which I think makes sense.